R/visualize.R

```diff
@@ -11,30 +11,44 @@
 #' @param dens_color A character or hex string specifying the color of the
 #'   theoretical density curve.
 #' @param obs_stat A numeric value or 1x1 data frame corresponding to what the
-#'   observed statistic is.
+#'   observed statistic is. **Deprecated (see Details)**.
 #' @param obs_stat_color A character or hex string specifying the color of the
-#'   observed statistic as a vertical line on the plot.
+#'   observed statistic as a vertical line on the plot. **Deprecated (see
+#'   Details)**.
 #' @param pvalue_fill A character or hex string specifying the color to shade
 #'   the p-value. In previous versions of the package this was the `shade_color`
-#'   argument.
+#'   argument. **Deprecated (see Details)**.
 #' @param direction A string specifying in which direction the shading should
 #'   occur. Options are `"less"`, `"greater"`, or `"two_sided"` for p-value. Can
 #'   also give `"left"`, `"right"`, or `"both"` for p-value. For confidence
 #'   intervals, use `"between"` and give the endpoint values in `endpoints`.
+#'   **Deprecated (see Details)**.
 #' @param endpoints A 2 element vector or a 1 x 2 data frame containing the
 #'   lower and upper values to be plotted. Most useful for visualizing
-#'   conference intervals.
+#'   conference intervals. **Deprecated (see Details)**.
 #' @param endpoints_color A character or hex string specifying the color of the
-#'   observed statistic as a vertical line on the plot.
+#'   observed statistic as a vertical line on the plot. **Deprecated (see
+#'   Details)**.
 #' @param ci_fill A character or hex string specifying the color to shade the
-#'   confidence interval.
-#' @param ... Other arguments passed along to ggplot2.
+#'   confidence interval. **Deprecated (see Details)**.
+#' @param ... Other arguments passed along to \\{ggplot2\\} functions.
+#'
+#' @details In order to make visualization workflow more straightforward and
+#' explicit `visualize()` now only should be used to plot statistics directly.
+#' That is why arguments not related to this task are deprecated and will be
+#' removed in a future release of \\{infer\\}.
+#' 
+#' To add to plot information related to p-value use [shade_p_value()]. To add
+#' to plot information related to confidence interval use
+#' [shade_confidence_interval()].
```

Comment:
This is a great design improvment!

Reply:
Yeah, it unfortunates will cause some old code to eventually break but I think it's better.

Reply:
I'm not exactly sure how much `visualize()` is used in the courses. I'm happy to assist with converting them over as needed. Just let me know.

Reply:
All previous `visualize()` code will work in this version but with warnings if `obs_stat` or `endpoints` arguments are used.